### PR TITLE
[#1112] Fix query plan caching for Hibernate and add config option to…

### LINKED ISSUE #1112: Invalid use of hibernate query plan cache when participating queries use limit/offset

### PR DESCRIPTION
**Pull request by Mobe91**
… enable/disable query plan caching

fixes #1112

### CODE REVIEW

File: core/api/src/main/java/com/blazebit/persistence/ConfigurationProperties.java [hunk 1]
```diff
@@ -179,7 +179,19 @@
...
+     * If set to true, the query plans are cached and reused.
+     * Valid values for this property are <code>true</code> and <code>false</code>.
+     * Default is <code>true</code>.
+     * This configuration option currently only takes effect when Hibernate is used as JPA provider.
+     *
+     * The property can be changed for a criteria builder before constructing a query.
+     *
+     * @since 1.5.0
+     */
+    public static final String QUERY_PLAN_CACHE_ENABLED = "com.blazebit.persistence.query_plan_cache_enabled";
```
Comment (jwgmeligmeyling) [inline core/api/src/main/java/com/blazebit/persistence/ConfigurationProperties.java:193]: Isn't query caching opt-in anyway through `setCacheable(true)`?

Comment (Mobe91) [reply core/api/src/main/java/com/blazebit/persistence/ConfigurationProperties.java:193]: `setCacheable(true)` affects query result caching.

File: core/impl/src/main/java/com/blazebit/persistence/impl/AbstractCTECriteriaBuilder.java [hunk 2]
```diff
@@ -118,9 +118,11 @@ protected Query getQuery(String baseQueryString) {
             // The main query will handle that separately
             if (!isMainQuery) {
                 if (firstResult != 0) {
+                    query.setFirstResult(firstResult);
                     offset = Integer.toString(firstResult);
                 }
                 if (maxResults != Integer.MAX_VALUE) {
+                    query.setMaxResults(maxResults);
```
[the remark this answers is not in the document]
Comment (Mobe91) [reply core/impl/src/main/java/com/blazebit/persistence/impl/AbstractCTECriteriaBuilder.java:125]: This is part of the actual fix. The other part is the extension of the query plan cache key to also consider limit/offset of participating queries.

File: core/api/src/main/java/com/blazebit/persistence/ConfigurationProperties.java (hunk 1 above)
Comment (jwgmeligmeyling) [reply core/api/src/main/java/com/blazebit/persistence/ConfigurationProperties.java:193]: Ah! My bad!

**Comment by Mobe91** [comment]
Closed in favor of #1115

**Pull request closed without merging**